lib/live_head.ex

```diff
@@ -164,7 +164,7 @@ defmodule Phx.Live.Head do
     #  head events or we replace the whole list of events with our mapped variant
     #  including merged head events
     if merged? do
-      put_in(socket.private.__changed__.push_events, events)
+      put_in(socket.private.__temp__.push_events, events)
```

Comment:
I think we can make the fix without needing a version bump by detecting which field is supported. Could you have a try with something like:

```elixir
cond do
  # Phoenix LiveView < 0.19
  merged? && Map.has_key?(socket.private, :__changed__) ->
    put_in(socket.private.__changed__.push_events, events)
  
  # Phoenix LiveView >= 0.19
  merged? -> 
    put_in(socket.private.__temp__.push_events, events)
    
  true -> 
    push_event(socket, "hd", %{c: [[query, [change]]]})
end
```